src/sentry/tagstore/snuba/backend.py

```diff
@@ -380,6 +384,126 @@ def get_groups_user_counts(self, project_id, group_ids, environment_id):
                              referrer='tagstore.get_groups_user_counts')
         return defaultdict(int, {k: v for k, v in result.items() if v})
 
+    def get_tag_value_paginator(self, project_id, environment_id, key, query=None,
+            order_by='-last_seen'):
+        from sentry.api.paginator import SequencePaginator
+
+        if not order_by == '-last_seen':
+            raise ValueError("Unsupported order_by: %s" % order_by)
+
+        conditions = []
+        if query:
+            conditions.append(['tags_value', 'LIKE', '%{}%'.format(query)])
+
+        start, end = self.get_time_range()
+        results = snuba.query(
+            start=start,
+            end=end,
+            groupby=['tags_value'],
+            filter_keys={
+                'project_id': [project_id],
+                'environment': [environment_id],
+                'tags_key': [key],
+            },
+            aggregations=[
+                ['count()', '', 'times_seen'],
+                ['min', 'timestamp', 'first_seen'],
+                ['max', 'timestamp', 'last_seen'],
+            ],
+            conditions=conditions,
+            orderby=order_by,
+            # TODO: This means they can't actually paginate all TagValues.
+            limit=1000,
+        )
+
+        tag_values = [
+            TagValue(
+                key=key,
+                value=k or u'',
+                times_seen=v['times_seen'],
+                first_seen=parse_date(v['first_seen']),
+                last_seen=parse_date(v['last_seen']),
+            ) for k, v in six.iteritems(results)
+        ]
+
+        desc = order_by.startswith('-')
+        score_field = order_by.lstrip('-')
+        return SequencePaginator(
+            [(time.mktime(getattr(tv, score_field).timetuple()), tv) for tv in tag_values],
```

Comment:
I think you can use `sentry.utils.dates.to_timestamp` here instead if you truncate it to an integer afterwards. Also, if this is going to be consistent with the other time-based cursors, this would need to be multiplied by 1000:

https://github.com/getsentry/sentry/blob/ef6b12996c5814ff737154a1a793474f090c7859/src/sentry/search/snuba/backend.py#L55-L60